### PR TITLE
Feature: Allow User to Hide a Pinned Post

### PR DESCRIPTION
**Pull request by JonathanPorta**
- Ensure that a user can hide pinned posts.
- Fix a bug causing the pinned post icon to be highlighted constantly.
- Clean up a couple of style issues caught by the linter.

### Shortcomings
The values that determine whether to hide the pinned post and whether to highlight the pinned icon are saved in the browser's local storage. Therefore, these changes do not persist to other devices.

> image
> Small Screen (Expanded) - Dismiss Icon + Highlighted Pin Icon (First Visit)

>image
> Small Screen (Expanded) - Pinned Post Dismissed

>image
> Small Screen (Expanded) - Dismiss Icon + Highlighted Pin Icon (Subsequent Visit)

Replaces https://github.com/steemit/condenser/pull/3265 - Branch Naming Issue

### CODE REVIEW

File: src/app/redux/UserSaga.js [hunk 1]
```diff
@@ -635,9 +636,21 @@ function* uploadImage({
     };
     xhr.upload.onprogress = function(event) {
         if (event.lengthComputable) {
-            const percent = Math.round(event.loaded / event.total * 100);
+            const percent = Math.round(event.loadedx / event.total * 100);
```
Comment (rjungemann) [inline src/app/redux/UserSaga.js]: Might be a typo?

File: src/app/redux/UserSaga.js [hunk 2]
```diff
@@ -635,9 +636,21 @@ function* uploadImage({
...
     };
     xhr.send(formData);
 }
+
+function* hidePinnedPost(action) {
+    console.log('userSaga hidePinnedPost', action);
+    const closedPinnedPosts = yield select(state =>
+        state.user.get('closedPinnedPosts')
+    );
+    console.log('userSaga closedPinnedPosts', closedPinnedPosts);
```
Comment (rjungemann) [inline src/app/redux/UserSaga.js]: If you want to keep the console calls around, change them to `console.info` if you don't mind

File: src/app/redux/GlobalReducer.js [hunk 3]
```diff
@@ -120,7 +120,7 @@ export default function reducer(state = defaultState, action = {}) {
             return payload.pinnedPosts.reduce((acc, pinnedPost) => {
                 const author = pinnedPost.get('author');
                 const permlink = pinnedPost.get('permlink');
-                return state.updateIn(
+                return acc.updateIn(
```
Comment (rjungemann) [inline src/app/redux/GlobalReducer.js]: This fixes a bug with marking pinned posts as seen. Good catch!

**Comment by rjungemann** [comment]
Jonathan and I did discuss using sagas for local storage, but since it's synchronous, since it's high-performance, and since we're calling `localStorage` in views elsewhere, we agreed that this functionality didn't need to live in the saga.

**Comment by rjungemann** [comment]
I'll let Tim or Justin do the honors of reviewing!

File: src/app/redux/GlobalReducer.js (hunk 3 above)
Comment (JonathanPorta) [reply src/app/redux/GlobalReducer.js]: Thanks for the help!

File: src/app/redux/UserSaga.js (hunk 1 above)
Comment (JonathanPorta) [reply src/app/redux/UserSaga.js]: Thanks for catching this!

**Comment by roadscape** [comment]
Tested, LGTM!

**Pull request merged**